monai/apps/generation/maisi/networks/autoencoderkl_maisi.py

```diff
@@ -0,0 +1,929 @@
+        dropout_dim: int = 1,
+        dilation: Sequence[int] | int = 1,
+        groups: int = 1,
+        bias: bool = True,
+        conv_only: bool = False,
+        is_transposed: bool = False,
+        padding: Sequence[int] | int | None = None,
+        output_padding: Sequence[int] | int | None = None,
+    ) -> None:
+        super().__init__()
+        self.conv = Convolution(
+            spatial_dims=spatial_dims,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            strides=strides,
+            kernel_size=kernel_size,
+            adn_ordering=adn_ordering,
+            act=act,
+            norm=norm,
+            dropout=dropout,
+            dropout_dim=dropout_dim,
+            dilation=dilation,
+            groups=groups,
+            bias=bias,
+            conv_only=conv_only,
+            is_transposed=is_transposed,
+            padding=padding,
+            output_padding=output_padding,
+        )
+
+        self.dim_split = dim_split
+        self.stride = strides[self.dim_split] if isinstance(strides, list) else strides
+        self.num_splits = num_splits
+        self.debug = debug
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        num_splits = self.num_splits
+        if self.debug:
+            print("num_splits:", num_splits)
+
+        l = x.size(self.dim_split + 2)
+        split_size = l // num_splits
+
+        padding = 3
+        if padding % self.stride > 0:
+            padding = (padding // self.stride + 1) * self.stride
+        if self.debug:
+            print("padding:", padding)
+
+        overlaps = [0] + [padding] * (num_splits - 1)
+        last_padding = x.size(self.dim_split + 2) % split_size
+
+        if self.dim_split == 0:
+            splits = [
+                x[
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                    :,
+                    :,
+                ]
+                for i in range(num_splits)
+            ]
+        elif self.dim_split == 1:
+            splits = [
+                x[
+                    :,
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                    :,
+                ]
+                for i in range(num_splits)
+            ]
+        elif self.dim_split == 2:
+            splits = [
+                x[
+                    :,
+                    :,
+                    :,
+                    :,
+                    i * split_size
+                    - overlaps[i] : (i + 1) * split_size
+                    + (padding if i != num_splits - 1 else last_padding),
+                ]
+                for i in range(num_splits)
+            ]
```

Comment:
```suggestion
        slices = [slice(None)] * 5
        slices[self.dim_split + 2] = slice(i * split_size - overlaps[i], (i + 1) * split_size + (padding if i != num_splits - 1 else last_padding)]
        slices = x[tuple(slices)]
```

Reply:
Thanks! @ericspod I refined the script based on your suggestion.

Reply:
Hi @ericspod , I have updated the script. Shall we resolve the thread? Thanks.